### ISSUE TITLE
Error handler in extensions not called anymore since 2.0.9

### ISSUE THREAD

**Issue by frigus02** [async-graphql/async-graphql#336]
## Expected Behavior

When my query/mutation handler returns a `FieldResult::Err`, the `error` handler in extensions such as `Tracing` should be called.

## Actual Behavior

Since version 2.0.9 the `error` handler is not called.

## Steps to Reproduce the Problem

The following example prints "Error: something went wrong" to the console for version 2.0.8, but prints nothing for versions 2.0.9-2.0.13:

```rust
use async_graphql::extensions::{Extension, ExtensionContext, ExtensionFactory};
use async_graphql::{
    EmptyMutation, EmptySubscription, FieldResult, Object, Schema, ServerError, ID,
};

pub struct Tracing;

impl ExtensionFactory for Tracing {
    fn create(&self) -> Box<dyn Extension> {
        Box::new(TracingExtension)
    }
}

struct TracingExtension;

impl Extension for TracingExtension {
    fn error(&mut self, _ctx: &ExtensionContext<'_>, err: &ServerError) {
        println!("Error: {}", err);
    }
}

struct Query;

#[Object]
impl Query {
    async fn get_fruit(&self, _id: ID) -> FieldResult<String> {
        Err("something went wrong".into())
    }
}

#[async_std::main]
async fn main() {
    let schema = Schema::build(Query, EmptyMutation, EmptySubscription)
        .extension(Tracing)
        .finish();
    schema.execute("{ getFruit(id: \"apple\") }").await;
}
```

## Specifications

- Version: 2.0.9 - 2.0.13
- Platform: macOS

**Comment by sunli829** [comment]
Released `v2.0.14`.

**Comment by frigus02** [comment]
Can confirm, that it works as expected. Thanks for the quick fix.
